### ISSUE TITLE
Provide an easy way to have multiline descriptions in add-ons

### ISSUE THREAD

**Issue by jorgemoralespou** [minishift/minishift#1136]
When having an addon with a long description, you need to write it as a single line:

e.g.:

````
# Name: prometheus
# Description: This template creates a Prometheus instance preconfigured to gather OpenShift and Kubernetes platform and node metrics and report them to admins. It is protected by an OAuth proxy that only allows access for users who have view access to the prometheus namespace. You may customize where the images (built from openshift/prometheus and openshift/oauth-proxy) are pulled from via template parameters.
````

It would be nice to allow for better readability with something that allowed multiline:

`````
# Name: prometheus
# Description: This template creates a Prometheus instance preconfigured to gather OpenShift and \
                       Kubernetes platform and node metrics and report them to admins. It is protected by an \
                       OAuth proxy that only allows access for users who have view access to the prometheus \
                       namespace. You may customize where the images (built from openshift/prometheus \
                       and openshift/oauth-proxy) are pulled from via template parameters.
````

NOTE: Notation is not even a recommendation ;-)

**Comment by hferentschik** [comment]
I had this on my todo list as well.

**Comment by hferentschik** [comment]
@praveenkumar The code is here https://github.com/minishift/minishift/blob/29b2959185e716056cf85ae5c6b81c6d354ad8c4/pkg/minishift/addon/parser/addon_parser.go#L182-L182

As far as I can tell, all header lines prefixed with '#' gets passed to _createMetaData_. It then created a map of meta tag to meta value. This is using a regexp for now which works on a line by line basis. This would need to change. How depends on how we want to support multi lines. Do we require a '\' at the end to indicate continuation? Or do we want to have additional indentation on the following lines? Or nothing of this and all lines following a "Foo: bla bla" will be "merged" into the "Foo" meta until either a new tag appears of the end of header is reached. 

Up to us how to change this.

**Comment by gbraad** [comment]
You suggest to implement this as a state machine? Where state changes when
a different recognizable instruction has been detected.

-- 

   Gerard Braad | http://gbraad.nl
   [ Doing Open Source Matters ]

**Comment by hferentschik** [comment]
Pretty much. Start iterating the lines. If you detect a meta tag start collecting all following lines within this metatag until you reach a new meta tag or hit another condition which wold indicate that the current meta tag data is complete (for example the line is not indented anymore)

**Comment by hferentschik** [comment]
Merged via pull request #1280
